.github/workflows/medusa.yml

```diff
@@ -0,0 +1,184 @@
+name: Run Medusa tests
```

Comment:
Can we refactor this into a bash script that can be run locally to reproduce CI failures and just run the bash script in CI?

Reply:
Yes, but IMHO what we need is to have an medusa action here. I think @elopez can draft one when he is around, so this code will be reduced and have same shape as the echidna one.